
# HG changeset patch
# User welpy-cw <h.w.forms@arcor.de>
# Date 1766333226 -3600
# Node ID 9a774b8d91fb3407e90dba8f7b5b5079cd65703a
# Parent  d319ee132d01a17f6c81e30a7a0d09d614f53746
Bug 1938799 - Indicate new messages when creating a folder row. r=mkmelin

Folder rows only indicated new messages by turning blue when new messages were
received or when the folder was collapsed or expanded. Newly created rows for
folders containing new messages did not. For this reason, folders that had been
read and were added to the unread view after receiving new messages missed the
indication. The same occurred for all folders when a new three-pane tab was
opened, or folder modes were switched off and on.

Differential Revision: https://phabricator.services.mozilla.com/D276098

diff --git a/mail/base/content/widgets/folder-tree-row.mjs b/mail/base/content/widgets/folder-tree-row.mjs
--- a/mail/base/content/widgets/folder-tree-row.mjs
+++ b/mail/base/content/widgets/folder-tree-row.mjs
@@ -299,8 +299,14 @@ class FolderTreeRow extends HTMLLIElemen
     this._folderName = folder.abbreviatedName;
     this._fullFolderName = folder.prettyPath;
     this._setName();
     const isCollapsed = this.classList.contains("collapsed");
+    this.classList.toggle(
+      "new-messages",
+      isCollapsed
+        ? folder.hasFolderOrSubfolderNewMessages
+        : folder.hasNewMessages
+    );
     this.unreadCount = folder.getNumUnread(isCollapsed);
     this.totalCount = folder.getTotalMessages(isCollapsed);
     if (lazy.XULStoreUtils.isItemVisible("messenger", "folderPaneFolderSize")) {
       this.folderSize = this.formatFolderSize(folder.sizeOnDisk);
